### PR TITLE
[CIVIS-2934] skippable tests api client

### PR DESCRIPTION
**Pull request by anmarchenko**
**What does this PR do?**
Add skippable tests api client: will be used in subsequent PRs to request from the backend which tests can be skipped

**How to test the change?**
Unit tests only for now

### CODE REVIEW

File: lib/datadog/ci/itr/skippable.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        class Test
+          attr_reader :name, :suite
+
+          def initialize(name:, suite:)
+            @name = name
+            @suite = suite
+          end
+
+          def ==(other)
+            name == other.name && suite == other.suite
```
Comment (romainkomorndatadog) [inline lib/datadog/ci/itr/skippable.rb:21]: Are you missing the bundle (which is the module) here?

File: lib/datadog/ci/itr/skippable.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          {
+            "data" => {
+              "type" => Ext::Transport::DD_API_SKIPPABLE_TESTS_TYPE,
+              "attributes" => {
+                "test_level" => Ext::Test::ITR_TEST_SKIPPING_MODE,
+                "service" => test_session.service,
+                "env" => @dd_env,
+                "repository_url" => test_session.git_repository_url,
+                "sha" => test_session.git_commit_sha,
+                "configurations" => {
```
Comment (romainkomorndatadog) [inline lib/datadog/ci/itr/skippable.rb:100]: Do you plan on adding support for custom configurations in a separate PR?

File: lib/datadog/ci/itr/skippable.rb (hunk 1 above)
Comment (anmarchenko) [reply lib/datadog/ci/itr/skippable.rb:21]: ruby library does not use modules

Comment (anmarchenko) [reply lib/datadog/ci/itr/skippable.rb:21]: so I just use the same approach as JS library: only take name and suite from the payload

File: lib/datadog/ci/itr/skippable.rb (hunk 2 above)
Comment (anmarchenko) [reply lib/datadog/ci/itr/skippable.rb:100]: yes, I will: https://datadoghq.atlassian.net/browse/CIVIS-9333

File: lib/datadog/ci/itr/skippable.rb (hunk 1 above)
Comment (romainkomorndatadog) [reply lib/datadog/ci/itr/skippable.rb:21]: Interesting!

**Review by romainkomorndatadog — APPROVED** [review]

**Pull request merged**